src/components/modal/index.tsx

```diff
@@ -1,44 +1,51 @@
 import _ from 'lodash';
-import PropTypes from 'prop-types';
-import React from 'react';
-import {StyleSheet, Modal as RNModal, TouchableWithoutFeedback} from 'react-native';
+import React, {Component} from 'react';
+import {StyleSheet, Modal as RNModal, ModalProps, TouchableWithoutFeedback, GestureResponderEvent} from 'react-native';
 import {BlurView} from '@react-native-community/blur';
 import {Constants} from '../../helpers';
-import {BaseComponent} from '../../commons';
-import TopBar from './TopBar';
+import {asBaseComponent} from '../../commons/new';
+import TopBar, {ModalTopBarPropTypes} from './TopBar';
```

Comment:
rename `ModalTopBarPropTypes` to `ModalTopBarProps` 
(see comment below)

Reply:
Done